### PR TITLE
Add documentation workflow

### PR DESCRIPTION
**Pull request by 1e-to**
Closes #286
Closes #275

### CODE REVIEW

File: .github/workflows/generate-docs.yml [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    branches:
+      - main
+      - release*
+    tags:
+      - '[0-9]+.[0-9]+*'
+
+env:
+  PACKAGE_NAME: numba-dppy
+  python: 3.8
+  artifact_name: -c dppy_label_dev
```
Comment (PokhodenkoSA) [inline .github/workflows/generate-docs.yml:13]: It is not necessary for docs pipeline.

File: .github/workflows/generate-docs.yml [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  dependencies: dpnp=0.7.1=*_41
+  CHANNELS: -c dppy/label/dev -c defaults -c numba -c intel -c numba/label/dev -c conda-forge --override-channels
```
Comment (PokhodenkoSA) [inline .github/workflows/generate-docs.yml:15]: Remove `-c dppy/label/dev` from highest priority.
```suggestion
  CHANNELS: -c defaults -c numba -c intel -c numba/label/dev -c dppy/label/dev -c conda-forge --override-channels
```

File: .github/workflows/generate-docs.yml [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+          conda build \
+            $TEST \
+            $VERSIONS \
+            $CHANNELS \
+            conda-recipe
+      - name: Upload artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ env.PACKAGE_NAME }} ${{ runner.os }} Python ${{ env.python }} ${{ env.artifact_name }}
+          path: /usr/share/miniconda/conda-bld/linux-64/${{ env.PACKAGE_NAME }}-*.tar.bz2
```
Comment (PokhodenkoSA) [inline .github/workflows/generate-docs.yml:57]: It is ok for docs pipeline to eliminate splitting to build and test jobs.

Comment (PokhodenkoSA) [reply .github/workflows/generate-docs.yml:57]: So upload/download artifacts is not necessary.

File: .github/workflows/generate-docs.yml [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+          # Test channel
+          conda search $PACKAGE_NAME -c $GITHUB_WORKSPACE/channel --override-channels
+      - name: Install numba-dppy
+        run: |
+          CHANNELS="-c $GITHUB_WORKSPACE/channel ${{ env.CHANNELS }}"
+          conda install $PACKAGE_NAME python=${{ env.python }} ${{ env.dependencies }} $CHANNELS
+          # Test installed packages
+          conda list
+      - name: Add library
+        run: echo "OCL_ICD_FILENAMES=libintelocl.so" >> $GITHUB_ENV
```
Comment (PokhodenkoSA) [inline .github/workflows/generate-docs.yml:100]: Is it necessary w/o tests?

**Review by PokhodenkoSA — APPROVED** [review]
LGTM. But there is proposals for improvements.

**Comment by PokhodenkoSA** [comment]
Fix pre-commit, please.

File: .github/workflows/generate-docs.yml (hunk 4 above)
Comment (1e-to) [reply .github/workflows/generate-docs.yml:63]: Yes, it is the main problem why module doesn't import

**Comment by PokhodenkoSA** [comment]
@1e-to, please, fix CI.

**Comment by PokhodenkoSA** [comment]
@1e-to, internal error fixed. Please, restart CI.

**Pull request merged**